lib/models/project.js

```diff
@@ -255,7 +263,24 @@ class Project {
       let appConfig = this.require(configPath)(env);
       let addonsConfig = this.getAddonsConfig(env, appConfig);
 
-      return _.merge(addonsConfig, appConfig);
+      const mergedConfig = _.merge(addonsConfig, appConfig);
+
+      const deprecationCompliance = {};
+      deprecationCompliance.self = this.pkg.ember.deprecationCompliance;
+      deprecationCompliance.addons = {};
+
+      this.addons.forEach((x) => {
+        if (x.pkg.ember && x.pkg.ember.deprecationCompliance) {
+          console.log('✅', 'deprecation compliance: ', x.name, x.pkg.version, x.pkg.ember.deprecationCompliance);
+          deprecationCompliance.addons[x.name] = x.pkg.ember.deprecationCompliance;
+        } else {
+          console.log('💩', 'deprecation compliance: ', x.name, x.pkg.version, );
+        }
+      });
+      mergedConfig.EmberENV.DEPRECATION_COMPLIANCE = deprecationCompliance;
```

Comment:
tried this out w/ a sample app and looks like `this.addons` is already the canonical list and handles nested dependencies, etc, but I'm not 100% sure.

Seems like just passing the whole thing into EmberEnv is ok? @rwjblue super initial pass, but let me know if you had anything else in mind.